File: java/src/main/java/defpackage/ScrollCaption.java
package defpackage;

import javax.microedition.lcdui.Image;

/* renamed from: bc */
/* JADX INFO: loaded from: Heroes-Lore-Wind-of-Soltia_J2ME_EN_v207.jar:ScrollCaption.class */
/**
 * One line of the end-credits staff roll: a pre-rendered caption {@link #image}
 * and its current vertical position {@link #y}. {@link GameScreen} spawns these
 * into its {@code creditCaptions} vector as the ending text advances, scrolls
 * each up two pixels per frame, and drops it once it leaves the top of the
 * screen.
 */
public final class ScrollCaption {
    /* renamed from: a */
    /** Pre-rendered caption image for this credits line. */
    public Image image;

    /* JADX INFO: renamed from: a, reason: collision with other field name */
    /** Current y position (screen pixels); decremented as the roll scrolls up. */
    public int y;

    public ScrollCaption(Image image, int y) {
        this.image = image;
        this.y = y;
    }
}

File: transliteration/game-xlat/src/scroll_caption.rs
//! Transliterated from `java/src/main/java/defpackage/ScrollCaption.java`
//! (original `bc.class` in `Heroes-Lore-Wind-of-Soltia_J2ME_EN_v207.jar`).
//!
//! Implementation #1: strict transliteration. See `docs/TRANSLITERATION.md`.
//!
//! One line of the end-credits staff roll: a pre-rendered caption [`image`] plus its
//! current vertical [`y`]. `GameScreen` spawns these into its (DEFERRED)
//! `creditCaptions` vector as the ending text advances, scrolls each up two pixels
//! per frame, and drops it once it leaves the top of the screen. A leaf data holder
//! — no methods beyond the constructor.
//!
//! Opcode shapes (R8, `_reference/numeric_shapes.json`):
//! `bc.<init>:(Ljavax/microedition/lcdui/Image;I)V => []` (plain field stores).
//!
//! [`image`]: ScrollCaption::image
//! [`y`]: ScrollCaption::y

use j2me_me::Image;

/// Java `bc` / `ScrollCaption` — one credits-roll line: a pre-rendered caption image
/// and its scroll position. Owned per-instance by `GameScreen`'s (deferred)
/// `creditCaptions` vector; no statics.
#[derive(Debug, Default)]
pub struct ScrollCaption {
    /// `public Image image;` (`bc.a`) — the pre-rendered caption bitmap.
    /// `None` == Java `null` (the reference-nullable model, as `MapObject.image`).
    pub image: Option<Image>,
    /// `public int y;` (`bc.a`, `I`) — current y position (screen pixels),
    /// decremented as the roll scrolls up.
    pub y: i32,
}

impl ScrollCaption {
    /// `public ScrollCaption(Image image, int y)`
    /// (`bc.<init>:(Ljavax/microedition/lcdui/Image;I)V => []`):
    /// `this.image = image; this.y = y;`.
    pub fn new(image: Option<Image>, y: i32) -> Self {
        // this.image = image;
        // this.y = y;
        ScrollCaption { image, y }
    }
}
